src/deepsparse/transformers/engines/nl_decoder_engine.py

```diff
@@ -256,13 +261,27 @@ def add_kv_cache_to_input(self, inp: List[numpy.ndarray]) -> List[numpy.ndarray]
         """
         Takes the input and adds the past kv cache state to it.
 
+        If the internal kv cache support is enabled, the kv cache state
+        will always be reinitialized to zeros. This is just to make sure
```

Comment:
can try `empty` instead of zeros

Reply:
https://stackoverflow.com/questions/26213199/np-fullsize-0-vs-np-zerossize-vs-np-empty

Ironically, zeros is faster than empty.